segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -301,7 +304,8 @@ private void handleAppendResult(final Append append, Throwable exception, Timer
                     latestEventNumbers.put(Pair.of(append.getSegment(), append.getWriterId()), append.getEventNumber());
                 } else {
                     if (!conditionalFailed) {
-                        waitingAppends.removeAll(append.getWriterId());
+                        pendingBytes -= waitingAppends.removeAll(append.getWriterId())
```

Comment:
You still haven't answered my original question. What does `.readableBytes()` return there? The buffer will have already been read, so it's likely this is 0.

There is no unit test that verifies this value. You need to write one.

Reply:
@andreipaduroiu 
yes, right now, it always returns 0 only; but , however, if the conditional append exists in between appends and the conditional failed, then it should be removed. hence code in 307 is required.
The existing tests in the AppendProcessorTest.java covering the 100% code l but I could not able to add any test case to induce the conditional append in between the data appends, because  Append processor append method invokes performNextWrite and performNextWrite invokes getNextAppend which reads the appends and all these calls are synchronous. 

one more thing about this PR is not just avoiding the loop; I want to this pending bytes to metrics in near future as a separate PR.

Reply:
For the test you can add an additional append in the middle by creating two append processors that share the same store.

Reply:
@tkaitchuck 
   I tried to do this; but , problem is this flow : performNextWrite -> getNextAppend  which always clears the pending bytes. please give me some guide lines here.
However , now, I have added additional code ensure that pendingBytes values never reaches negative value.

Reply:
Here's an idea:
- Make a protected method in `AppendDecoder` which sets the value of `pendingBytes`
- Subclass `AppendDecoder` and override that method
- In the subclass, record all values assigned to this field in an ordered list
- In your test, verify that the values recorded in this list are as you expected.

Reply:
Hi @andreipaduroiu 
      AppendDecoder? but the 'pendingBytes" are local to "AppendProcessor".
I have added an private method getPendingBytes;  this can made as protected; however, this values is always turnns out 0 because Appendprocessor's Append method calls performNextWrite -> getNextAppend always clears this to 0.

Reply:
My bad. I was probably thinking of another PR when I wrote that comment. I meant `AppendProcessor`.

To answer your question, the way you would clear that value is by setting it to 0 (or subtracting until you reach 0). What I was suggesting was that, instead of directly assigning a new value to that field, we create a protected method that sets the field to whatever value we pass as an argument. We change all callsites to invoke this method instead of directly assigning it. Then in your unit test, you can subclass AppendProcessor and override the method so that you can record every single value it had.

Reply:
Thanks a lot @andreipaduroiu 
yes,  I will post the unit test as per your above review comments.

Reply:
Hi @andreipaduroiu 
    I have added the unit tests ; please let me know your further review comments.